internal/website/content/howto/pubsub/publish.md

```diff
@@ -0,0 +1,410 @@
+---
+title: "Publish Messages to a Topic"
+date: 2019-03-26T09:44:15-07:00
+draft: true
+weight: 1
+---
+
+Publishing a message to a topic with the Go CDK takes two steps:
+
+1. Open the topic with the Pub/Sub provider of your choice (once per topic).
+2. Send messages on the topic.
+
+The second step is the same across all providers because the first step
+creates a value of the portable [`*pubsub.Topic`][] type. Publishing looks
+like this:
+
+```go
+err := topic.Send(ctx, &pubsub.Message{
+    Body: []byte("Hello, World!\n")
+    Metadata: map[string]string{
+        "Foo": "Bar",
+    },
+})
+if err != nil {
+    return err
+}
+```
+
+The rest of this guide will discuss how to accomplish the first step: opening
+a topic for your chosen Pub/Sub provider.
+
+[`*pubsub.Topic`]: https://godoc.org/gocloud.dev/pubsub#Topic
+
+## Constructors versus URL openers
+
+If you know that your program is always going to use a particular Pub/Sub
```

Comment:
There's a longer non-API-specific discussion of URLs here:
https://godoc.org/gocloud.dev#hdr-URLs

Consider linking to it?

Reply:
I'd like to move that information to a Concepts page as per #1469 and then I'd link to it. I could link to that section in the interrim tho, WDYT?